src/main/scala/org/zalando/spearheads/innkeeper/services/PathsService.scala

```diff
@@ -57,17 +58,40 @@ class DefaultPathsService @Inject() (pathsRepo: PathsRepo)(implicit val executio
 
     pathsRepo.pathWithUriHostIdExists(path.uri, path.hostIds)
       .flatMap {
-        case false => pathsRepo.insert(pathRow).flatMap(rowToEventualMaybePath)
-        case true  => Future.successful(Failure(DuplicatePathUriHost()))
+        case false =>
+          val insertResult = pathsRepo.insert(pathRow)
+
+          insertResult.onSuccess {
+            case insertedPath => insertedPath.id.foreach { id =>
+              auditsRepo.persistPathLog(id, createdBy.name, AuditType.Create)
+            }
+          }
+
+          insertResult.flatMap(rowToEventualMaybePath)
+
+        case true =>
+          Future.successful(Failure(DuplicatePathUriHost()))
       }
   }
 
   override def patch(
     id: Long,
     path: PathPatch,
+    userName: String,
     updatedAt: LocalDateTime): Future[ServiceResult.Result[PathOut]] = {
 
-    pathsRepo.update(id, path, updatedAt).flatMap {
+    val updateResult = pathsRepo.update(id, path, updatedAt)
+
+    updateResult.onSuccess {
```

Comment:
I would extract all of the `onSuccess` callbacks out of the main service methods (in other private methods, or even another class), as they are another concern (auditing), different of the main one (doing some business logic).